src/core/nixl_agent.cpp

```diff
@@ -762,6 +762,39 @@ nixlAgent::createXferReq(const nixl_xfer_op_t &operation,
     return NIXL_SUCCESS;
 }
 
+nixl_status_t
+nixlAgent::estimateXferCost(const nixlXferReqH *req_hndl,
+                            std::chrono::microseconds &duration,
+                            std::chrono::microseconds &err_margin,
+                            nixl_cost_estimate_t &source,
+                            const nixl_opt_args_t* extra_params) const
+{
+    NIXL_LOCK_GUARD(data->lock);
```

Comment:
should it be `NIXL_SHARED_LOCK_GUARD(data->lock);` now?
(note it was merged recently)